yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala

```diff
@@ -211,15 +211,15 @@ private[yarn] class ExecutorRunnable(
       Seq("--user-class-path", "file:" + absPath)
     }.toSeq
 
+    // Kill if OOM is raised - leverage yarn's failure handling to cause rescheduling.
+    // Not killing the task leaves various aspects of the executor and (to some extent) the jvm in
+    // an inconsistent state.
+    // TODO: If the OOM is not recoverable by rescheduling it on different node, then do
+    // 'something' to fail job ... akin to blacklisting trackers in mapred ?
```

Comment:
Really this comment belongs with the method doc -- should have been there before. You could move it